lib/web/fetch/formdata-parser.js

```diff
@@ -290,34 +286,61 @@ function parseMultipartFormDataHeaders (input, position) {
         //    failure.
         name = parseMultipartFormDataName(input, position)
 
-        if (name === null) {
-          return 'failure'
-        }
-
         // 5. If position points to a sequence of bytes starting with `; filename="`:
-        if (bufferStartsWith(input, filenameBuffer, position)) {
-          // Note: undici also handles filename*
-          let check = position.position + filenameBuffer.length
-
-          if (input[check] === 0x2a) {
-            position.position += 1
-            check += 1
-          }
-
-          if (input[check] !== 0x3d || input[check + 1] !== 0x22) { // ="
-            return 'failure'
-          }
-
-          // 1. Advance position so it points at the byte after the next 0x22 (") byte
-          //    (the one in the sequence of bytes matched above).
-          position.position += 12
-
-          // 2. Set filename to the result of parsing a multipart/form-data name given
-          //    input and position, if the result is not failure. Otherwise, return failure.
-          filename = parseMultipartFormDataName(input, position)
-
-          if (filename === null) {
-            return 'failure'
+        if (input[position.position] === 0x3b /* ; */ && input[position.position + 1] === 0x20 /* ' ' */) {
+          const at = { position: position.position + 2 }
+
+          if (bufferStartsWith(input, filenameBuffer, at)) {
+            if (input[at.position + 8] === 0x2a /* '*' */) {
+              at.position += 10 // skip past filename*=
+
+              // Remove leading http tab and spaces. See RFC for examples.
+              // https://datatracker.ietf.org/doc/html/rfc6266#section-5
+              collectASequenceOfBytes(
+                (char) => char === 0x20 || char === 0x09,
+                input,
+                at
+              )
+
+              const headerValue = collectASequenceOfBytes(
+                (char) => char !== 0x20 && char !== 0x0d && char !== 0x0a, // ' ' or CRLF
+                input,
+                at
+              )
+
+              if (
+                (headerValue[0] !== 0x75 && headerValue[0] !== 0x55) || // u or U
+                (headerValue[1] !== 0x74 && headerValue[1] !== 0x54) || // t or T
+                (headerValue[2] !== 0x66 && headerValue[2] !== 0x46) || // f or F
+                headerValue[3] !== 0x2d || // -
+                headerValue[4] !== 0x38 // 8
+              ) {
+                throw parsingError('unknown encoding, expected utf-8\'\'')
```

Comment:
I have no idea if `utf-8` is the only accepted value here. Could not find references to other encodings in the RFC.